backend/x11/backend.c

```diff
@@ -231,20 +229,58 @@ static bool wlr_x11_backend_start(struct wlr_backend *backend) {
 		return false;
 	}
 
-	init_atom(x11, &x11->atoms.wm_protocols, 1, "WM_PROTOCOLS");
-	init_atom(x11, &x11->atoms.wm_delete_window, 0, "WM_DELETE_WINDOW");
-	init_atom(x11, &x11->atoms.net_wm_name, 1, "_NET_WM_NAME");
-	init_atom(x11, &x11->atoms.utf8_string, 0, "UTF8_STRING");
+	struct {
+		const char *name;
+		bool only_if_exists;
+		xcb_intern_atom_cookie_t cookie;
+		xcb_atom_t *atom;
+	} atom[] = {
+		{
+			.name = "WM_PROTOCOLS",
+			.only_if_exists = true,
+			.atom = &x11->atoms.wm_protocols,
+		},
+		{
+			.name = "WM_DELETE_WINDOW",
+			.only_if_exists = false,
+			.atom = &x11->atoms.wm_delete_window,
+		},
+		{
+			.name = "_NET_WM_NAME",
+			.only_if_exists = true,
+			.atom = &x11->atoms.net_wm_name,
+		},
+		{
+			.name = "UTF8_STRING",
+			.only_if_exists = false,
+			.atom = &x11->atoms.utf8_string,
+		},
+	};
```

Comment:
Unrelated to this commit, but: Why is `only_if_exists` not just always set to `true`?

Reply:
Most of this code was copied from some pretty poor examples. It was probably there originally as some portability to some ancient X server that nobody uses any more.